### ISSUE TITLE
Config file, where do I get it from?

### ISSUE THREAD

**Issue by zell-mbc** [chrishrb/solarlog-exporter#6]
When starting up with docker-compose solarlog-exporter crashes immediately, complaining about "No inverters in config found", which is probably correct because I did not provide a config file.
The readme says:

1. Get data from LOCAL
2. Parse Config file

What does this config file look like and where should it come from? An export from SolarLog?

**Comment by chrishrb** [comment]
yes, the config file is in the directory where all the other solarlog files are stored. its name is `base_vars.js`. It provides some infos about the whole system.

**Comment by zell-mbc** [comment]
Thanks @chrishrb, that file is indeed in the folder with all the other js files.

I forgot to add the data volume in my docker-compose file.
The config error is gone but now I am stuck with 

`solarlog-exporter  | requests.exceptions.ConnectionError: HTTPConnectionPool(host='solarlog-db', port=8086): Max retries exceeded with url: /query?q=SELECT+%2A+FROM+solarlog_min+WHERE+system+%3D+%27PV-Anlage%27+ORDER+BY+time+DESC+LIMIT+1%3B&db=solarlog (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fa99a7cae50>: Failed to establish a new connection: [Errno 111] Connection refused'))
`

Looks like the DB was found but it couldn't authorize? I'll look into this some more over the weekend.
Like they say, failing in different ways is progress :-)

**Comment by zell-mbc** [comment]
Hi @chrishrb, after leaving this dormant for a while I wanted to pick things up again. And I made some progress in such that I now get stuck with 
`solarlog-exporter  | ERROR:root:No inverters in config found!
`

Looking at the code (parser.py, #124) I can see that you are looking for a section labeled "AnlagenGrp". Now I don't have that section in my config. Maybe this is because my Solarlog 500 is pretty old and the format of the config changed over time?

```
if "AnlagenGrp" not in self._config:
            return None

```            

Here's the relevant section from my config file:
```
$ cat data/base_vars.js
var Boot=99
var AnlagenKWP=14820
var time_start=new Array(8,8,6,6,6,6,6,7,7,7,7,8)
var time_end=new Array(17,18,20,21,21,22,22,21,20,19,17,17)
var sollMonth=new Array(2,6,9,11,11,13,13,12,10,6,4,3)
var SollYearKWP=890
var AnzahlWR=4
var MaxWRP=new Array(4)
MaxWRP[0]=new Array(4600,35000,650000,5000000)
MaxWRP[1]=new Array(4000,30000,600000,4500000)
MaxWRP[2]=new Array(3900,30000,550000,4500000)
MaxWRP[3]=new Array(3900,30000,550000,4500000)
var WRInfo=new Array(4)
WRInfo[0]=new Array("WR4_5TL-20","2100137561",4140,1,"WR1 Links",2,null,null,0,null,163,0,0,938,null,null,null,1)
WRInfo[0][6]=new Array("String 1")
WRInfo[0][7]=new Array(1)
WRInfo[0][9]=new Array(4140)
WRInfo[1]=new Array("WR4_5TL-20","2100138509",3680,1,"WR2 Rechts",2,null,null,0,null,163,0,0,938,null,null,null,1)
WRInfo[1][6]=new Array("")
WRInfo[1][7]=new Array(1)
WRInfo[1][9]=new Array(3680)
WRInfo[2]=new Array("WR25-W03"," 854323084",3500,2,"WR3 Unten",1,null,null,0,null,1,0,0,995,null,null,null,1)
WRInfo[2][6]=new Array("")
WRInfo[2][7]=new Array(2)
WRInfo[2][9]=new Array(3500)
WRInfo[3]=new Array("WR25-008"," 854423160",3500,2,"WR4 Oben",1,null,null,0,null,1,0,0,995,null,null,null,1)
WRInfo[3][6]=new Array("")
WRInfo[3][7]=new Array(2)
WRInfo[3][9]=new Array(3500)
var HPTitel="MBC"
var HPBetreiber=""
…
```

**Comment by zell-mbc** [comment]
Wondering if you maybe have some time to have a look at this?
In the meantime I managed to run the python app outside of Docker to be able to debug more easily.
As you can see your code is plastered with debug prints() to help me understand what's going on, but to be honest my head is spinning right now and I thought I'd pick it up with you again.

Up-left is the code section where things break right now
Bottom left, the debug output
Right, my base_vars.js

- The print() in 137 produces this output "Default_config: [4.0, [['WR4_5TL-20…", a nested string/array with all 4 inverter settings
- 139 shows the value item to be = "4.0" while I think this should be an array as otherwise the append in 140 doesn't make sense? - Line 140 is where things break anyway.
- And even if I somehow would manage to get over this, things would fail in get-group (line 124) because I don't have groups in my config.

Now, I was considering to somehow hardcoding the inverter settings, bypassing all the enumerations, but I worry that this may be pointless because my config may be fundamentally different from what you expect? 

![image](https://user-images.githubusercontent.com/29018767/199317767-94c49623-9936-4ab6-b026-787dab81e21b.png)

**Comment by chrishrb** [comment]
hm thats weird.. the only difference in the `base_vars` that i notice, is in your config the `new Array(4)`. Try to replace it with `new Array(AnzahlWR)` in both lines:

```js
...
var MaxWRP=new Array(AnzahlWR)
...
var WRInfo=new Array(AnzahlWR)
..

```

I tried your config, an that hack did it for me. Maybe not the best solution, but yea it works.. parsing js with python wasn't the best idea - maybe one day i rewrite this script..

**Comment by zell-mbc** [comment]
Boy, I have been battling this all morning…
First of all, your proposal worked, but what made things hard for me is the lack of knowledge of InfluxDB.

I started off with a current release (2.x) of InfluxDB. Had to change the logon method from user / password to token in core.py to be able to connect to the db. But this only worked when testing with running the python app manually, the moment I reverted to the Dockerfile build, authentication  failed. Probably different support library versions pulled somewhere…couldn't figure it out and eventually reverted to a InfluxDB < 2.0 and username/pwd authentication.

On the positive side, Grafana is connected and shows proper values! Now I need to figure out the best way to retrieve the Solarlog updates…almost there :-)

![image](https://user-images.githubusercontent.com/29018767/199477872-d76a2d01-48c4-454f-9650-93fdf6fe5806.png)
